docs/content/guide/filter.ngdoc

```diff
@@ -109,7 +109,7 @@ as the first argument. Any filter arguments are passed in as additional argument
 function.
 
 The filter function should be a [pure function](http://en.wikipedia.org/wiki/Pure_function), which
-means that it should be stateless and idempotent, and not rely for example on other Angular services.
+means that it should be deterministic and not rely, for example, on other Angular services.
```

Comment:
> idempotent is not actually correct for filters.

Here, it is used in the broader, "computer science" meaning:
_"In computer science, the term idempotent is used more comprehensively to describe an operation that will produce the same results if executed once or multiple times."_

But I agree it can be confusing, given the stricter mathematical (and pure functional programming) meanings.

--
> `deterministic` actually does cover the idea of `stateless` since all stateless functions are inherently deterministic.

This is not necessarily true (depending on the strictness of your definition). Basically, `stateless` isn't accurate either.

--
Since we already have a link to formal definition of "pure function", why don't we keep it simple here and change it to something like:

```diff
 The filter function should be a [pure function](http://en.wikipedia.org/wiki/Pure_function), which
-means that it should be stateless and idempotent, and not rely for example on other Angular services.
+means that it should always return the same result given the same input arguments and should not affect external state, for example, other Angular services.
```